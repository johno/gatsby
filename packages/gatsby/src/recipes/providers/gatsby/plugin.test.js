const fs = require(`fs-extra`)
const path = require(`path`)

const plugin = require(`./plugin`)
const { addPluginToConfig, getPluginsFromConfig } = require(`./plugin`)
const resourceTestHelper = require(`../resource-test-helper`)

const root = path.join(__dirname, `./fixtures`)
const name = `gatsby-plugin-foo`
const options = { from: `future` }
const configPath = path.join(root, `gatsby-config.js`)

describe(`gatsby-plugin resource`, () => {
  test(`e2e plugin resource test`, async () => {
    await resourceTestHelper({
      resourceModule: plugin,
      resourceName: `GatsbyPlugin`,
      context: { root },
      initialObject: { id: name, name },
      partialUpdate: { id: name, options },
    })
  })

  test(`does not add the same plugin twice by default`, async () => {
    const configSrc = await fs.readFile(configPath, `utf8`)
    const newConfigSrc = addPluginToConfig(
      configSrc,
      `gatsby-plugin-react-helmet`
    )
    const plugins = getPluginsFromConfig(newConfigSrc)

    const result = [...new Set(plugins)]

    expect(result).toEqual(plugins)
  })

  test(`all returns an array of plugins`, async () => {
    const result = await plugin.all({ root })

    expect(result).toMatchSnapshot()
  })
})
