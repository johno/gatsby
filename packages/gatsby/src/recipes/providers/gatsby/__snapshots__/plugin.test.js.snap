// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin resource all returns an array of plugins 1`] = `
Array [
  Object {
    "id": "\`gatsby-source-filesystem\`",
    "name": "\`gatsby-source-filesystem\`",
    "options": Object {
      "name": null,
      "path": null,
    },
  },
  Object {
    "id": "\`gatsby-transformer-sharp\`",
    "name": "\`gatsby-transformer-sharp\`",
    "options": Object {},
  },
  Object {
    "id": "\`gatsby-plugin-emotion\`",
    "name": "\`gatsby-plugin-emotion\`",
    "options": Object {},
  },
  Object {
    "id": "\`gatsby-plugin-typography\`",
    "name": "\`gatsby-plugin-typography\`",
    "options": Object {
      "pathToConfigModule": null,
    },
  },
  Object {
    "id": "\`gatsby-transformer-remark\`",
    "name": "\`gatsby-transformer-remark\`",
    "options": Object {
      "plugins": null,
    },
  },
  Object {
    "id": "\`gatsby-plugin-sharp\`",
    "name": "\`gatsby-plugin-sharp\`",
    "options": Object {},
  },
  Object {
    "id": "\`gatsby-plugin-google-analytics\`",
    "name": "\`gatsby-plugin-google-analytics\`",
    "options": Object {
      "trackingId": null,
    },
  },
  Object {
    "id": "\`gatsby-plugin-manifest\`",
    "name": "\`gatsby-plugin-manifest\`",
    "options": Object {
      "background_color": null,
      "display": null,
      "icon": null,
      "name": null,
      "short_name": null,
      "start_url": null,
      "theme_color": null,
    },
  },
  Object {
    "id": "\`gatsby-plugin-offline\`",
    "name": "\`gatsby-plugin-offline\`",
    "options": Object {},
  },
  Object {
    "id": "\`gatsby-plugin-react-helmet\`",
    "name": "\`gatsby-plugin-react-helmet\`",
    "options": Object {},
  },
]
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin create 1`] = `
Object {
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "options": Object {},
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin create plan 1`] = `
Object {
  "currentState": "const redish = \`#c5484d\`;
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`
  },
  plugins: [{
    resolve: \`gatsby-source-filesystem\`,
    options: {
      path: \`\${__dirname}/src/pages\`,
      name: \`pages\`
    }
  }, \`gatsby-transformer-sharp\`, \`gatsby-plugin-emotion\`, {
    resolve: \`gatsby-plugin-typography\`,
    options: {
      pathToConfigModule: \`src/utils/typography\`
    }
  }, {
    resolve: \`gatsby-transformer-remark\`,
    options: {
      plugins: [{
        resolve: \`gatsby-remark-images\`,
        options: {
          maxWidth: 590
        }
      }, {
        resolve: \`gatsby-remark-responsive-iframe\`,
        options: {
          wrapperStyle: \`margin-bottom: 1.0725rem\`
        }
      }, \`gatsby-remark-prismjs\`, \`gatsby-remark-copy-linked-files\`, \`gatsby-remark-smartypants\`]
    }
  }, \`gatsby-plugin-sharp\`, {
    resolve: \`gatsby-plugin-google-analytics\`,
    options: {
      trackingId: \`UA-774017-3\`
    }
  }, {
    resolve: \`gatsby-plugin-manifest\`,
    options: {
      name: \`Bricolage\`,
      short_name: \`Bricolage\`,
      icon: \`static/logo.png\`,
      start_url: \`/\`,
      background_color: redish,
      theme_color: redish,
      display: \`minimal-ui\`
    }
  }, \`gatsby-plugin-offline\`, \`gatsby-plugin-react-helmet\`]
};",
  "describe": "Configure gatsby-plugin-foo",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "newState": "const redish = \`#c5484d\`;
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`
  },
  plugins: [{
    resolve: \`gatsby-source-filesystem\`,
    options: {
      path: \`\${__dirname}/src/pages\`,
      name: \`pages\`
    }
  }, \`gatsby-transformer-sharp\`, \`gatsby-plugin-emotion\`, {
    resolve: \`gatsby-plugin-typography\`,
    options: {
      pathToConfigModule: \`src/utils/typography\`
    }
  }, {
    resolve: \`gatsby-transformer-remark\`,
    options: {
      plugins: [{
        resolve: \`gatsby-remark-images\`,
        options: {
          maxWidth: 590
        }
      }, {
        resolve: \`gatsby-remark-responsive-iframe\`,
        options: {
          wrapperStyle: \`margin-bottom: 1.0725rem\`
        }
      }, \`gatsby-remark-prismjs\`, \`gatsby-remark-copy-linked-files\`, \`gatsby-remark-smartypants\`]
    }
  }, \`gatsby-plugin-sharp\`, {
    resolve: \`gatsby-plugin-google-analytics\`,
    options: {
      trackingId: \`UA-774017-3\`
    }
  }, {
    resolve: \`gatsby-plugin-manifest\`,
    options: {
      name: \`Bricolage\`,
      short_name: \`Bricolage\`,
      icon: \`static/logo.png\`,
      start_url: \`/\`,
      background_color: redish,
      theme_color: redish,
      display: \`minimal-ui\`
    }
  }, \`gatsby-plugin-offline\`, \`gatsby-plugin-react-helmet\`, \\"gatsby-plugin-foo\\"]
};",
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin destroy 1`] = `undefined`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin update 1`] = `
Object {
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "options": Object {},
}
`;

exports[`gatsby-plugin resource e2e plugin resource test: GatsbyPlugin update plan 1`] = `
Object {
  "currentState": "const redish = \`#c5484d\`;
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`
  },
  plugins: [{
    resolve: \`gatsby-source-filesystem\`,
    options: {
      path: \`\${__dirname}/src/pages\`,
      name: \`pages\`
    }
  }, \`gatsby-transformer-sharp\`, \`gatsby-plugin-emotion\`, {
    resolve: \`gatsby-plugin-typography\`,
    options: {
      pathToConfigModule: \`src/utils/typography\`
    }
  }, {
    resolve: \`gatsby-transformer-remark\`,
    options: {
      plugins: [{
        resolve: \`gatsby-remark-images\`,
        options: {
          maxWidth: 590
        }
      }, {
        resolve: \`gatsby-remark-responsive-iframe\`,
        options: {
          wrapperStyle: \`margin-bottom: 1.0725rem\`
        }
      }, \`gatsby-remark-prismjs\`, \`gatsby-remark-copy-linked-files\`, \`gatsby-remark-smartypants\`]
    }
  }, \`gatsby-plugin-sharp\`, {
    resolve: \`gatsby-plugin-google-analytics\`,
    options: {
      trackingId: \`UA-774017-3\`
    }
  }, {
    resolve: \`gatsby-plugin-manifest\`,
    options: {
      name: \`Bricolage\`,
      short_name: \`Bricolage\`,
      icon: \`static/logo.png\`,
      start_url: \`/\`,
      background_color: redish,
      theme_color: redish,
      display: \`minimal-ui\`
    }
  }, \`gatsby-plugin-offline\`, \`gatsby-plugin-react-helmet\`, \\"gatsby-plugin-foo\\"]
};",
  "describe": "Configure gatsby-plugin-foo",
  "id": "gatsby-plugin-foo",
  "name": "gatsby-plugin-foo",
  "newState": "const redish = \`#c5484d\`;
module.exports = {
  siteMetadata: {
    title: \`Bricolage\`,
    author: \`Kyle Mathews\`,
    homeCity: \`San Francisco\`
  },
  plugins: [{
    resolve: \`gatsby-source-filesystem\`,
    options: {
      path: \`\${__dirname}/src/pages\`,
      name: \`pages\`
    }
  }, \`gatsby-transformer-sharp\`, \`gatsby-plugin-emotion\`, {
    resolve: \`gatsby-plugin-typography\`,
    options: {
      pathToConfigModule: \`src/utils/typography\`
    }
  }, {
    resolve: \`gatsby-transformer-remark\`,
    options: {
      plugins: [{
        resolve: \`gatsby-remark-images\`,
        options: {
          maxWidth: 590
        }
      }, {
        resolve: \`gatsby-remark-responsive-iframe\`,
        options: {
          wrapperStyle: \`margin-bottom: 1.0725rem\`
        }
      }, \`gatsby-remark-prismjs\`, \`gatsby-remark-copy-linked-files\`, \`gatsby-remark-smartypants\`]
    }
  }, \`gatsby-plugin-sharp\`, {
    resolve: \`gatsby-plugin-google-analytics\`,
    options: {
      trackingId: \`UA-774017-3\`
    }
  }, {
    resolve: \`gatsby-plugin-manifest\`,
    options: {
      name: \`Bricolage\`,
      short_name: \`Bricolage\`,
      icon: \`static/logo.png\`,
      start_url: \`/\`,
      background_color: redish,
      theme_color: redish,
      display: \`minimal-ui\`
    }
  }, \`gatsby-plugin-offline\`, \`gatsby-plugin-react-helmet\`, \\"gatsby-plugin-foo\\"]
};",
}
`;
